### DISCUSSION TITLE
Dual mirror

### DISCUSSION THREAD

**Discussion by jumpjack** [ricktu288/ray-optics#132] [Ideas]
How do I simulate a mirror which reflects on one side but lets light pass throigh grom the other side?

**Comment by ricktu288** [comment]
I think that is not how one-way mirror works (see https://phydemo.app/ray-optics/gallery/interrogation-room).

Of course there can be idealized unrealistic objects in the simulator (also for #131), but can you give some examples on how it can be useful (also for #131)?

**Reply by jumpjack** [reply]
I would like to know how much of the received light is reflected by a mirror.

`{ "version": 2, "objs": [ { "type": "parabolicmirror", "p1": { "type": 1, "x": 90.00833892822266, "y": 266.4474792480469, "exist": true }, "p2": { "type": 1, "x": 254.0471954345703, "y": 268.0465393066406, "exist": true }, "p3": { "type": 1, "x": 172.27621030671588, "y": 241.76058307012806, "exist": true } }, { "type": "led", "p1": { "type": 1, "x": 244.99307250976562, "y": 430.07861328125, "exist": true }, "p2": { "type": 1, "x": 157.11512756347656, "y": 194.4924774169922, "exist": true }, "p": 36.001, "symmetric": true }, { "type": "blackline", "p1": { "type": 1, "x": 61.780860900878906, "y": 284.56951904296875, "exist": true }, "p2": { "type": 1, "x": 312.0998840332031, "y": 285.10247802734375, "exist": true } } ], "mode": "light", "rayDensity_light": 0.1, "rayDensity_images": 1, "showGrid": false, "grid": false, "lockobjs": false, "gridSize": 20, "observer": null, "origin": { "x": 0, "y": 0 }, "scale": 1, "width": 400, "height": 800, "colorMode": false, "symbolicGrin": false }`

![Screenshot_20240324_120330_Chrome.jpg](https://github.com/ricktu288/ray-optics/assets/1620953/62a60c3c-8a3c-4bf8-b5b7-1c9a8b8a1c99)

![Screenshot_20240324_120354_Chrome.jpg](https://github.com/ricktu288/ray-optics/assets/1620953/9a8d6767-24e9-429b-931e-d30c665a35a4)

**Comment by ricktu288** [comment]
Now it is possible to use a single-sided Tools -> Other -> Custom Surface
[Example](https://phydemo.app/ray-optics/simulator/#XQAAAALUAAAAAAAAAABCKcrGU8hqLFnpmU9EnTFrVldP-XiP9Y4WtYuOC-ZDaTnwCj1gZMfSiIkSNXCfWnlz3h4hqJHwS5Jc4GBi9hqfCEgspN98mHJtJYNLTt9edAnSM3cvpC6bymF_IacnJMMhy5v0GtDipFll2FSjzrKeMWvKK0UdFnG0tRNnSDOpUzoMUkK8YvlKBtuG673mIArNtEdiAOIg7Xs-___UdDAA)
